### DISCUSSION TITLE
Previous durables variables created in a flow are being deleted after a For Each Block is used in the flow

### DISCUSSION THREAD

**Discussion by jjruizruiz** [postmanlabs/postman-flows#182] [Defect]
Hello,

I was trying to create a flow and I have durables created while the flow is running, after a ForEach block is used, durables not related with the iterated response where the ForEach block is linked are deleted.

Could you please check it and help me?

Thank you so much in advance

**Comment by Manuel-delablanca** [comment]
Hi,

I'm having exactly the same problem. My Durables are lost after I use a for each block.

Thanks!

**Comment by therubioftw** [comment]
+1, same problem.
It's weird, because when I set up the next block, it recognizes that there are durables in the flow, and I'm able to set them up...
BUT
...when I start the flow, the durables don't get through.

**Reply by tomalazarov** [reply]
Exactly the same... I've made durable the claims_token which I need for further requests and it is working in the requests before the For Each block but after it, on the last request.. the durable is visible and I can assign it to claims_token variable in the request... but at the end it's not working. In the console I can see that it actually sends {{CLAIMS_TOKEN}} instead of the actual value. 

Please fix it.

**Reply by saswatds** [reply]
I was able to reproduce this. Will get it fixed as part of the next release.
